src/Module/Search/Acl.php

```diff
@@ -0,0 +1,374 @@
+				];
+				if ($entry['forum']) {
+					$forums[] = $entry;
+				} else {
+					$contacts[] = $entry;
+				}
+			}
+			if (count($forums) > 0) {
+				if ($search == '') {
+					$forums[] = ['separator' => true];
+				}
+				$contacts = array_merge($forums, $contacts);
+			}
+		}
+
+		$items = array_merge($groups, $contacts);
+
+		if ($conv_id) {
+			// In multi threaded posts the conv_id is not the parent of the whole thread
+			$parent_item = Item::selectFirst(['parent'], ['id' => $conv_id]);
+			if (DBA::isResult($parent_item)) {
+				$conv_id = $parent_item['parent'];
+			}
+
+			/*
+			 * if $conv_id is set, get unknown contacts in thread
+			 * but first get known contacts url to filter them out
+			 */
+			$known_contacts = array_map(function ($i) {
+				return $i['link'];
+			}, $contacts);
+
+			$unknown_contacts = [];
+
+			$condition = ["`parent` = ?", $conv_id];
+			$params = ['order' => ['author-name' => true]];
+			$authors = Item::selectForUser(local_user(), ['author-link'], $condition, $params);
+			$item_authors = [];
+			while ($author = Item::fetch($authors)) {
+				$item_authors[$author['author-link']] = $author['author-link'];
+			}
+			DBA::close($authors);
+
+			foreach ($item_authors as $author) {
+				if (in_array($author, $known_contacts)) {
+					continue;
+				}
+
+				$contact = Contact::getDetailsByURL($author);
+
+				if (count($contact) > 0) {
+					$unknown_contacts[] = [
+						'type'    => 'c',
+						'photo'   => ProxyUtils::proxifyUrl($contact['micro'], false, ProxyUtils::SIZE_MICRO),
+						'name'    => htmlspecialchars($contact['name']),
+						'id'      => intval($contact['cid']),
+						'network' => $contact['network'],
+						'link'    => $contact['url'],
+						'nick'    => htmlentities(defaults($contact, 'nick', $contact['addr'])),
+						'addr'    => htmlentities(defaults($contact, 'addr', $contact['url'])),
+						'forum'   => $contact['forum']
+					];
+				}
+			}
+
+			$items = array_merge($items, $unknown_contacts);
+			$tot += count($unknown_contacts);
+		}
+
+		$results = [
+			'tot'      => $tot,
+			'start'    => $start,
+			'count'    => $count,
+			'groups'   => $groups,
+			'contacts' => $contacts,
+			'items'    => $items,
+			'type'     => $type,
+			'search'   => $search,
+		];
+
+		Hook::callAll('acl_lookup_end', $results);
+
+		$o = [
+			'tot'   => $results['tot'],
+			'start' => $results['start'],
+			'count' => $results['count'],
+			'items' => $results['items'],
+		];
+
+		echo json_encode($o);
+		exit;
+	}
+
+
+	/**
+	 * Searching for global contacts for autocompletion
+	 *
+	 * @brief Searching for global contacts for autocompletion
+	 * @param string $search Name or part of a name or nick
+	 * @param string $mode   Search mode (e.g. "community")
+	 * @param int    $page   Page number (starts at 1)
+	 * @return array with the search results
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	private static function contactAutocomplete($search, $mode, int $page = 1)
```

Comment:
This isn't a module call (interacting directly with the UI). It's a model call and it has no further dependency to this class, so I'd prefer moving this method to something like `Core\Search::searchGlobalContact`. In a near future, I'd like to move the ACL search logic to a `Search::acl...()` too and the module is just responsible for calling the right methods.

It's like:
- Model: `Search` class
- View: the template
- Controlller: the `Module` classes

Reply:
Fair enough.